### PR TITLE
Bug/release 2 task event bugs

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### Change description ###
Adding task event bug fixes in Release

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by udaydenduluri33 — DISMISSED** [review]

File: src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class CaseEventCompletionTaskReassignedComponent implements OnInit, OnDes
   public ngOnInit(): void {
     // Get case id and task from the parent component
     this.caseId = this.parentComponent.context.caseId;
-    const task = this.parentComponent.context.task;
+    const task = this.parentComponent.context.reassignedTask;
```
Comment (scott-turner-hmcts) [inline src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts:35]: very minor but you could use destructuring here to neaten things up:

```
const { caseId, reassignedTask } = this.parentComponent.context;
this.caseId = caseId;
const task = reassignedTask
```

File: src/shared/components/event-start/services/event-start-state-machine.service.ts [hunk 2]
```diff
@@ -197,9 +197,8 @@ export class EventStartStateMachineService {
   }
 
   public finalAction(state: State): void {
```
Comment (scott-turner-hmcts) [inline src/shared/components/event-start/services/event-start-state-machine.service.ts:199]: is this function even needed if it only does console.log or does it just need to be stubbed out for now?

**Review by scott-turner-hmcts — CHANGES_REQUESTED** [review]
couple of very minor comments to address otherwise looks good to me

File: src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts (hunk 1 above)
Comment (connorpgpmcelroy) [reply src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts:35]: Done

File: src/shared/components/event-start/services/event-start-state-machine.service.ts (hunk 2 above)
Comment (connorpgpmcelroy) [reply src/shared/components/event-start/services/event-start-state-machine.service.ts:199]: I'll comment out the log for now - think it was left in as precursor to other potential requirements

**Review by scott-turner-hmcts — DISMISSED** [review]

**Review by udaydenduluri33 — APPROVED** [review]
@connorpgpmcelroy  thanks for increasing the test coverage

File: src/shared/components/event-start/event-guard/event-start.guard.spec.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
-import { ActivatedRouteSnapshot } from '@angular/router';
+import { ActivatedRouteSnapshot, Router } from '@angular/router';
```
Comment (turner1979) [inline src/shared/components/event-start/event-guard/event-start.guard.spec.ts:1]: `Router` not in use here, can be removed

**Review by turner1979 — DISMISSED** [review]
Only one super minor suggestion from me, probably not even worth doing to be fair, so will approve

**Comment by connorpgpmcelroy** [comment]
Release 2 task event bugs no longer relevant as now finished release 4 without notable task event bugs

**Pull request closed without merging**